source/development_process/index.rst

```diff
@@ -0,0 +1,958 @@
+	  create_user true
+	  path node['whats_fresh']['application_dir']
+	  owner node['whats_fresh']['venv_owner']
+	  group node['whats_fresh']['venv_group']
+
+	  repository node['whats_fresh']['repository']
+	  revision node['whats_fresh']['git_branch']
+
+	  config_template 'config.yml.erb'
+	  config_destination "#{node['whats_fresh']['config_dir']}/config.yml"
+	  django_migrate true
+	  django_collectstatic true
+	  interpreter 'python2.7'
+
+	  gunicorn_port node['whats_fresh']['gunicorn_port']
+	end
+
+	nginx_app 'whats_fresh' do
+	  template 'whats_fresh.conf.erb'
+	  cookbook 'whats-fresh'
+	end
+
+This recipe uses several resources, including ``python_webapp``, ``nginx_app``,
+and ``directory``. The section ``node['whats_fresh']['application_dir']`` is
+just like a python dictionary with a specific configuration value, often a
+string.
+
+
+The Architecture of Chef, Abridged
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Chef has several parts, a central server which holds all of the cookbooks, and
+a collection of nodes, which are managed by the chef server. Each node has
+attributes which are essentially a hash or a dictionary of configuration values
+and keys. Each node has a list of roles it serves, like web server or database
+server, and a run list of recipes it will run. Every half hour or so each node
+checks with the chef server and converges to the state which the chef cookbook
+specifies. Sysadmins can edit the attributes of a node on the chef server using
+the ``knife`` tool.
+
+Anatomy of a Coobook
+~~~~~~~~~~~~~~~~~~~~
+Most cookbooks begin with a table of contents, but Chef cookbooks are full of
+files. Here is a summary of important files in a cookbook.
+
+.. code:: text
+
+	.
+	├── attributes: A directory
+	├── Berksfile: Defines other cookbooks on which this one depends
+	├── chefignore: Like a .gitignore, lists files which Chef won't upload.
+	├── Gemfile: Lists gem dependencies like python's requirements.txt
+	├── .kitchen.yml: Used for running Test Kitchen with Vagrant
+	├── .kitchen.cloud.yml: Used for running Test Kitchen on openstack
+	├── metadata.rb: Like the Berksfile, defines dependencies and cookbook info
+	├── providers: Providers define the code for a Light Weight Resource
+	│              Provider.
+	├── recipes: Recipes are ruby files instructing Chef how to configure a
+	│   │        node.
+	│   └── default.rb
+	├── resources: Resources define the interface of a Light Weight Resource
+	│              Provider.
+	├── templates: Holds Emedded Ruby (erb) templates for config files, etc.
+	│   └── default: Templates for the default recipe
+	├── test: Holds test kitchen tests.
+	│   └── integration
+	│       └── default: Tests for the default recipe
+	│           └── serverspec: Serverspec test files.
+	└── Vagrantfile: Used for starting Test Kitchen VMs
+
+
+Starting a New Cookbook
+~~~~~~~~~~~~~~~~~~~~~~~
+
+To begin a new project, run ``berks cookbook new-cookbook-name``. Berks will
```

Comment:
We have a [generator cookbook](https://github.com/osuosl-cookbooks/code_generator) that should be used with `berks cookbook`.